package.json

```diff
@@ -92,7 +92,7 @@
     "ethjs-util": "0.1.6",
     "keccak": "^3.0.0",
     "rlp": "^2.2.3",
-    "secp256k1": "^3.0.1"
+    "secp256k1": "^4.0.0"
```

Comment:
With this bump and keccak having been bumped to `^3.0.0` should an `"engines"` field be added to package.json?

```
"engines": { "node": ">=10.0.0" }
```

Likewise, should Node 8.x be dropped from the testing matrix?

Reply:
I have no experience with the `engines` field but I would assume it would make sense to use this throughout the EthereumJS libraries in general. Does this prevent an installation from happening when defined too strict?

Yes, Node 8 should then also be dropped from the test matrix. We nevertheless need to get these basic decisions outlined above done first.

Reply:
`"engines"` behavior varies depending on whether `npm` is used or `yarn` is used.

In the case of `npm`, the restriction is only enforced with respect to *your project's* `package.json`. That is, if dev/Deps listed in your project's `package.json` have an `"engines"` setting that isn't satisfied, it doesn't matter, they'll get installed anyway. But if your project's `package.json` has an `"engines"` setting that isn't satisfied, then running `npm install` in your project will fail.

In the case of `yarn` both your project's and its dependencies' `"engines"` settings are checked and `yarn install` will fail if there is any mismatch.

Either way, the `"engines"` setting is helpful metadata for humans, signaling the package manager's intention to support some Node versions and not others.

Reply:
Ok, thanks, but then I would assume at least to define a max version is somewhat dangerous and should eventually be skipped.

Reply:
Typically you only want a max version if you know that, for some reason, your package isn't compatible with a newer version of node.

For awhile the `embark` package had:

```
"engines": { "node": ">=10.17.0 <12.0.0" }
```

10.17.0 was the first Node version that had a change to Node's built-in scrypt that we wanted to be sure was in-place in a user's setup; at the same time, we had some transitive dependencies that weren't yet compatible with Node `>=12.0.0`. We eventually resolved the latter and now we have:

```
"engines": { "node": ">=10.17.0" }
```

Reply:
Yeah, right, likely a case-by-case decision. If there is a dependency clearly not compatible on the upper bound it makes sense to narrow. If one is is potentially - and unnecessarily - preventing users to upgrade to a future Node version, this should be left open.

Reply:
@cgewecke ah, and can you also add an `engines` field with `"engines": { "node": ">=10.0.0" }` as suggested in the first comment here?